logstash-core/lib/logstash/pipeline.rb

```diff
@@ -323,61 +385,64 @@ def flush_filters(options = {}, &block)
     end
   end
 
+  class FlusherObserver
+    def initialize(logger)
+      @logger = logger
+    end
+
+    def update(time, result, exception)
+      return unless exception
+      @logger.warn("Error during flush!",
+                   :message => exception.message,
+                   :class => exception.class.name,
+                   :backtrace => exception.backtrace)
+    end
```

Comment:
The only reason of timeout here, we are stuck on a push for `X TIMEOUT_DELAY`, I have looked at the TimerTask options / code they don't permit to create a task without a timeout. 

So in the worst case is the timer is schedule to run again and try to push the flush to the queue?
Which is similar to be block on the queue forever right? WDYT?